training/datastore/get_keywords.py

```diff
@@ -118,22 +118,27 @@ def from_github(self, package, url_df, api_url, api_token):
         else:
             return keywords
 
-    def find_keywords(self, df_, list_):
+    def find_keywords(self, data_, list_):
         """Find the keywords for given list of list of raw data."""
         package_lst = self.utility.flatten_list(list_)
         out_lst = list()
+        total = len(package_lst)
+        index = 0
         for i in package_lst:
-            pkg_kwd_lst = list()
-            pkg_kwd_lst = self.utility.make_list_from_series(
-                self.from_existing_df(df_, i))
-            if not pkg_kwd_lst or type(pkg_kwd_lst[2]) != list:
-                logger.info("Finding from the NPM repository.")
-                pkg_kwd_dict = self.from_npm_registry(i)
-                pkg_kwd_lst = list(pkg_kwd_dict.values())
-                if len(pkg_kwd_lst[2]) == 0:
-                    logger.info("Trying to fetch from Github")
-                    api_url = 'https://api.github.com/graphql'
-                    api_token = self.get_data.github_token
-                    pkg_kwd_lst[2] = self.from_github(i, df_, api_url, api_token)
+            index += 1
+            percentage = int(index * 100 / total)
+            logger.info(f'Processing [{index}/{total} => {percentage}%] => package {i}')
```

Comment:
If this is for tracking progress, can you print at an interval, this may produce too much log.

```suggestion
           if i%20 == 0:
                   logger.info(f'Processing [{index}/{total} => {percentage}%] => package {i}')
```

Reply:
Yes, this logs are only for tracking progress and also to know which package we are processing in the loop.
Agree to the point that there are too much logs, but this runs only one in a week as cron job. So, if something fails, investigating without complete information will become more complicated (like re-triggering entire pipeline to get more logs for investigation) . This was the reason I added logs and statistics very generously in this flows. In fact there are very few or no logs in current pipeline flow, which make it hard to figure issue when it occurs.

Reply:
@prashbnair This progress I have added to measure the time saved before and after our changes. I will remove this progress logs.